### ISSUE TITLE
[Story] Download file

### ISSUE THREAD

**Issue by BenoitTallandier** [linagora/Twake-Mobile#1143]
## User story summary
[Definition](#Definition)  
[Business case](#Business-case)  
[Criteria](#criteria)  
[UI/UX Design](#UI-UX-design)  
[Notes](#notes)  
[Related](#related)  

## Definition
```gherkin
Given I am a user
When I look to download a file from a discussion or a channel
Then I click on this file in the channel
And the file is downloaded in temporary folder
And the files open after it is download
And I click on the download button
When the download is completed, a simple image appears (for 1 second).
```

```gherkin
Given I am a user
When I look to download a image from a discussion or a channel
Then I click on this image in the channel
And a preview of it opens 
And I click on the download button
When the download is completed, a simple image appears (for 1 second).
```

```gherkin
Given I am a user
When I look to download a file from a channel
Then I click on the channel header
And I click on "files"
And I click on the file I want to download
And I click on the file when it is downloaded
And the file opens
And I click on the download button
When the download is completed, a simple image appears (for 1 second).
```

## Business case
- If a file is not already downloaded, there is the download icon in the attachment
- If a file is downloading, there is the loading icon in the attachment
- If a file is downloaded, there is the "file" icon in the attachment
- When I open an image from a channel, the preview show in which channel it was sent
- When I open an image from a direct discussion, the preview show the name of the file
- When I download an image or a video, it goes inside the gallery. Other files goes in the default folder.
- If the download is complete and I stayed on the file preview page, I can see a little image with a checkbox meaning that I well downloaded the file.

## Flow
![Group 61](https://user-images.githubusercontent.com/22451567/146960473-d0b660c8-7f09-4418-aa1b-73de871c3fc3.png)

## Criteria (Test case)

- [ ] When I download an image or a video, it goes inside the gallery. 
- [ ] When I download something else than an image or a video, it goes in the default folder.

## Related

**Comment by tlle14** [comment]
Hi @BenoitTallandier 
The story definition is different from the File Downloading demo.
Could you please review and update the story?

**Comment by BenoitTallandier** [comment]
Hi @tlle14 
I probably missed it, where do you see difference ? x)

**Comment by tlle14** [comment]
Hi @BenoitTallandier Kindly check my comment is in italic format.
**Case 1: File downloading**
Given I am a user
When I look to download a file from a discussion or a channel
Then I click on this file in the channel
And the file is downloaded in a temporary folder 
_=> As discussed with the Developer, I don't see the file is downloaded in a temporary folder._ 
And the files open after it is downloaded 
_=> Hence, some file types cannot be opened in preview_ 
And I click on the download button
When the download is completed, a simple image appears (for 1 second).
_=> It takes more than a second to download and display a button asking to open the file._ 

In conclusion:
- There is no temporary folder to download a file 
- There is no preview screen to preview files

**Comment by tlle14** [comment]
And @BenoitTallandier please confirm this behavior: 
With the attached image, you cannot download it on the Chat screen? 
As I see in the story and [Demo video](https://github.com/linagora/Twake-Mobile/pull/1160), the user needs to open previewing screen then download the image.
While on the design, it has a download button overlaying the image thumbnails. That means you can download the photo right on the chat screen. 
Which option we should implement?

**Comment by BenoitTallandier** [comment]
@tlle14 thank you for your explanation. I understand now :)

> Which option we should implement?

Now I watch the video, I think we should handle images like files. Preview is not good enough. I'll have the production version this week. I'll test it and I'll confirm it in the US :)

**Comment by tlle14** [comment]
That would be better. Thank you.
